### PR TITLE
fix(cli): quote Windows paths with spaces in Claude CLI validation

### PR DESCRIPTION
**Pull request by jeremykit**
When validating Claude CLI on Windows with shell: true (for .cmd/.bat files), paths containing spaces were being split by cmd.exe. For example: `D:\Program Files\nodejs\claude.cmd` was parsed as command `D:\Program` with arguments `Files\nodejs\claude.cmd --version`.

This fix adds proper quoting for paths with spaces when using shell: true, preventing cmd.exe from incorrectly splitting the path at space characters.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Base Branch

- [ ] This PR targets the `develop` branch (required for all feature/fix PRs)
- [ ] This PR targets `main` (hotfix only - maintainers)

## Related Issue

Closes #

## Type of Change

- [x] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 📚 Documentation
- [ ] ♻️ Refactor
- [ ] 🧪 Test

## Area

- [ ] Frontend
- [ ] Backend
- [ ] Fullstack

## Commit Message Format

Follow conventional commits: `<type>: <subject>`

**Types:** feat, fix, docs, style, refactor, test, chore

**Example:** `feat: add user authentication system`

## Checklist

- [ ] I've synced with `develop` branch
- [ ] I've tested my changes locally
- [ ] I've followed the code principles (SOLID, DRY, KISS)
- [ ] My PR is small and focused (< 400 lines ideally)

## CI/Testing Requirements

- [ ] All CI checks pass
- [ ] All existing tests pass
- [ ] New features include test coverage
- [ ] Bug fixes include regression tests

## Screenshots

| Before | After |
|--------|-------|
|        |       |

## Feature Toggle

- [ ] Behind localStorage flag: `use_feature_name`
- [ ] Behind settings toggle
- [ ] Behind environment variable/config
- [ ] N/A - Feature is complete and ready for all users

## Breaking Changes

**Breaking:** Yes / No

**Details:**

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed CLI validation on Windows so executable paths containing spaces are handled reliably during both synchronous and asynchronous checks. This reduces false failures when validating the local CLI and maintains existing invocation behavior and environment handling, improving consistency of detection and version-checking on Windows.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by AndyMik90 — CHANGES_REQUESTED** [review]
## 🤖 Auto Claude PR Review

### Merge Verdict: 🟠 NEEDS REVISION
1 issue(s) must be addressed (1 required, 0 recommended), 1 suggestions

### Risk Assessment
| Factor | Level | Notes |
|--------|-------|-------|
| Complexity | Low | Based on lines changed |
| Security Impact | None | Based on security findings |
| Scope Coherence | Good | Based on structural review |

### Findings Summary
- **High**: 1 issue(s)
- **Low**: 1 issue(s)

---
_Generated by Auto Claude PR Review_

### Findings (2 selected of 2 total)

#### 🟠 [7eea1a85ba75] [HIGH] Missing path quoting in async validateClaudeAsync() method
📁 `apps/frontend/src/main/cli-tool-manager.ts:1035`

The sync method validateClaude() (lines 928-930) was fixed to quote Windows paths with spaces when using shell: true, but the async counterpart validateClaudeAsync() was NOT updated with the same fix. When claudeCmd contains spaces (e.g., 'D:\Program Files\nodejs\claude.cmd') and needsShell is true, the async method will fail because cmd.exe splits the path at spaces. The async method is used by getToolPathAsync() which is called from the Electron main process.

**Suggested fix:**
```
Add the same quoting logic to validateClaudeAsync():

```typescript
private async validateClaudeAsync(claudeCmd: string): Promise<ToolValidation> {
  try {
    const needsShell = process.platform === 'win32' &&
      (claudeCmd.endsWith('.cmd') || claudeCmd.endsWith('.bat'));

    // When using shell: true on Windows, paths with spaces must be quoted
    // to prevent cmd.exe from splitting at spaces
    const cmdToRun = needsShell && claudeCmd.includes(' ')
      ? `"${claudeCmd}"`
      : claudeCmd;

    const { stdout } = await execFileAsync(cmdToRun, ['--version'], {
      ...
    });
```
```

#### 🔵 [523f33b93260] [LOW] Consider expanding quoting for other cmd.exe special characters
📁 `apps/frontend/src/main/cli-tool-manager.ts:928`

The fix only handles spaces, but Windows cmd.exe also interprets other special characters like &, |, >, <, ^, and % when using shell: true. For example, a path like 'D:\Test&Folder\claude.cmd' would cause cmd.exe to interpret & as a command separator. While this is an edge case for installation paths, expanding the check to include these characters would make the fix more robust.

**Suggested fix:**
```
Expand the condition to quote paths containing any special cmd.exe characters:

```typescript
const needsQuoting = needsShell && /[ &|<>^%]/.test(claudeCmd);
const cmdToRun = needsQuoting ? `"${claudeCmd}"` : claudeCmd;
```
```

---
*This review was generated by Auto Claude.*

File: apps/frontend/src/main/cli-tool-manager.ts [hunk 1]
```diff
@@ -923,7 +923,13 @@ class CLIToolManager {
       const needsShell = process.platform === 'win32' &&
         (claudeCmd.endsWith('.cmd') || claudeCmd.endsWith('.bat'));
 
-      const version = execFileSync(claudeCmd, ['--version'], {
+      // When using shell: true on Windows, paths with spaces must be quoted
+      // to prevent cmd.exe from splitting at spaces
+      const cmdToRun = needsShell && claudeCmd.includes(' ')
+        ? `"${claudeCmd}"`
+        : claudeCmd;
```
[the remark this answers is not in the document]
Comment (jeremykit) [reply apps/frontend/src/main/cli-tool-manager.ts]: ok

**Review by jeremykit — COMMENTED** [review]
accept

**Review by jeremykit — COMMENTED** [review]
ok

**Review by AndyMik90 — CHANGES_REQUESTED** [review]
## 🤖 Auto Claude PR Review

No new commits since last review. Previous findings still apply.

### Findings (2 selected of 2 total)

#### 🟠 [7eea1a85ba75] [HIGH] Missing path quoting in async validateClaudeAsync() method
📁 `apps/frontend/src/main/cli-tool-manager.ts:1035`

The sync method validateClaude() (lines 928-930) was fixed to quote Windows paths with spaces when using shell: true, but the async counterpart validateClaudeAsync() was NOT updated with the same fix. When claudeCmd contains spaces (e.g., 'D:\Program Files\nodejs\claude.cmd') and needsShell is true, the async method will fail because cmd.exe splits the path at spaces. The async method is used by getToolPathAsync() which is called from the Electron main process.

**Suggested fix:**
```
Add the same quoting logic to validateClaudeAsync():

```typescript
private async validateClaudeAsync(claudeCmd: string): Promise<ToolValidation> {
  try {
    const needsShell = process.platform === 'win32' &&
      (claudeCmd.endsWith('.cmd') || claudeCmd.endsWith('.bat'));

    // When using shell: true on Windows, paths with spaces must be quoted
    // to prevent cmd.exe from splitting at spaces
    const cmdToRun = needsShell && claudeCmd.includes(' ')
      ? `"${claudeCmd}"`
      : claudeCmd;

    const { stdout } = await execFileAsync(cmdToRun, ['--version'], {
      ...
    });
```
```

#### 🔵 [523f33b93260] [LOW] Consider expanding quoting for other cmd.exe special characters
📁 `apps/frontend/src/main/cli-tool-manager.ts:928`

The fix only handles spaces, but Windows cmd.exe also interprets other special characters like &, |, >, <, ^, and % when using shell: true. For example, a path like 'D:\Test&Folder\claude.cmd' would cause cmd.exe to interpret & as a command separator. While this is an edge case for installation paths, expanding the check to include these characters would make the fix more robust.

**Suggested fix:**
```
Expand the condition to quote paths containing any special cmd.exe characters:

```typescript
const needsQuoting = needsShell && /[ &|<>^%]/.test(claudeCmd);
const cmdToRun = needsQuoting ? `"${claudeCmd}"` : claudeCmd;
```
```

---
*This review was generated by Auto Claude.*

**Comment by jeremykit** [comment]
@AndyMik90 I've addressed the HIGH priority issue by adding the same quoting fix to `validateClaudeAsync()` in commit 4c6f3bc3. 

I've also synced my branch with the latest `develop`.

Could you please re-review when you have a chance? Thanks!

**Review by jeremykit — COMMENTED** [review]
OK

File: apps/frontend/src/main/cli-tool-manager.ts [hunk 2]
```diff
@@ -923,7 +923,13 @@ class CLIToolManager {
...
         (claudeCmd.endsWith('.cmd') || claudeCmd.endsWith('.bat'));
 
-      const version = execFileSync(claudeCmd, ['--version'], {
+      // When using shell: true on Windows, paths with spaces must be quoted
+      // to prevent cmd.exe from splitting at spaces
+      const cmdToRun = needsShell && claudeCmd.includes(' ') && !claudeCmd.startsWith('"')
+        ? `"${claudeCmd}"`
+        : claudeCmd;
+
+      const version = execFileSync(cmdToRun, ['--version'], {
```
[the remark this answers is not in the document]
Comment (jeremykit) [reply apps/frontend/src/main/cli-tool-manager.ts]: get it

File: apps/frontend/src/main/cli-tool-manager.ts [hunk 3]
```diff
@@ -1026,7 +1032,13 @@ class CLIToolManager {
...
         (claudeCmd.endsWith('.cmd') || claudeCmd.endsWith('.bat'));
 
-      const { stdout } = await execFileAsync(claudeCmd, ['--version'], {
+      // When using shell: true on Windows, paths with spaces must be quoted
+      // to prevent cmd.exe from splitting at spaces
+      const cmdToRun = needsShell && claudeCmd.includes(' ') && !claudeCmd.startsWith('"')
+      ? `"${claudeCmd}"`
+      : claudeCmd;
+
+      const { stdout } = await execFileAsync(cmdToRun, ['--version'], {
```
[the remark this answers is not in the document]
Comment (jeremykit) [reply apps/frontend/src/main/cli-tool-manager.ts]: get it

**Comment by AndyMik90** [comment]
📎 This PR appears to address Issue #1410 (Task fails to start because Claude Code not found on Windows). The path-with-spaces quoting fix could resolve cases where Windows paths cause CLI detection failures. Maintainers please verify.

**Comment by AndyMik90** [comment]
👋 Thanks so much for your interest in contributing to Aperant!

**Pull requests are currently paused.** We're rebuilding the entire app from the ground up for **Aperant 3.0** (including new cloud features). Most of that work is happening in a separate development repo right now and will be merged here once it's ready — so although this repo looks quiet, the project is very much alive. 🚀

Because the current codebase is being fully replaced, we're closing existing PRs so contributors aren't left waiting on changes that unfortunately can't be merged. This is **not** a reflection on your work, and we're sorry for any wasted effort. 🙏

We'd genuinely love your help shaping 3.0 instead:

- 🐛 **Report a bug:** https://github.com/AndyMik90/Aperant/issues
- 💡 **Share an idea / feature request:** https://github.com/AndyMik90/Aperant/discussions
- 💬 **Join the Discord:** https://discord.gg/KCXaPBr4Dj

Contributions will reopen once the 3.0 foundation lands. Thank you for understanding!
